src/main/java/uk/ac/cam/cl/dtg/isaac/quiz/PgQuestionAttempts.java

```diff
@@ -201,8 +201,15 @@ public void registerQuestionAttempt(final Long userId, final String questionPage
             } else {
                 pst.setNull(5, java.sql.Types.NULL);
             }
+
             pst.setTimestamp(6, new java.sql.Timestamp(questionAttempt.getDateAttempted().getTime()));
 
+            if (questionAttempt.getMarks() != null) {
+                pst.setInt(7, questionAttempt.getMarks());
+            } else {
+                pst.setInt(7, java.sql.Types.NULL);
```

Comment:
If `marks` is allowed to be `NULL` in the database, we have to be careful when reading it in below!